planner/core/logical_plan_builder.go

```diff
@@ -3803,7 +3803,15 @@ func (b *PlanBuilder) buildDataSource(ctx context.Context, tn *ast.TableName, as
 		dbName = model.NewCIStr(sessionVars.CurrentDB)
 	}
 
-	tbl, err := b.is.TableByName(dbName, tn.Name)
+	is := b.is
+	if len(b.buildingViewStack) > 0 {
+		if tempIs, ok := is.(*infoschema.TemporaryTableAttachedInfoSchema); ok {
+			// for tables in view, always ignore local temporary table
```

Comment:
Please give an example or describe it in more detail here. Others may be confused as I am: why do you need to check again in SELECT.